Source/SwiftLintFramework/Rules/LegacyCGGeometryFunctionsRule.swift

```diff
@@ -0,0 +1,155 @@
+                         "CGRectIsNull", "CGRectIsEmpty", "CGRectIsInfinite", "CGRectStandardize",
+                         "CGRectIntegral", "CGRectInset", "CGRectOffset", "CGRectUnion",
+                         "CGRectIntersection", "CGRectContainsRect", "CGRectContainsPoint",
+                         "CGRectIntersectsRect"]
+
+        let pattern = "\\b(" + functions.joinWithSeparator("|") + ")\\b"
+
+        return file.matchPattern(pattern, withSyntaxKinds: [.Identifier]).map {
+            StyleViolation(ruleDescription: self.dynamicType.description,
+                severity: configuration.severity,
+                location: Location(file: file, characterOffset: $0.location))
+        }
+    }
+
+    public func correctFile(file: File) -> [Correction] {
+        let varName = RegexHelpers.varNameGroup
+        let twoVars = RegexHelpers.twoVars
+        let twoVariableOrNumber = RegexHelpers.twoVariableOrNumber
+
+        let patterns = [
+            "CGRectGetWidth\\(\(varName)\\)": "$1.width",
+            "CGRectGetHeight\\(\(varName)\\)": "$1.height",
+            "CGRectGetMinX\\(\(varName)\\)": "$1.minX",
+            "CGRectGetMidX\\(\(varName)\\)": "$1.midX",
+            "CGRectGetMaxX\\(\(varName)\\)": "$1.maxX",
+            "CGRectGetMinY\\(\(varName)\\)": "$1.minY",
+            "CGRectGetMidY\\(\(varName)\\)": "$1.midY",
+            "CGRectGetMaxY\\(\(varName)\\)": "$1.maxY",
+            "CGRectIsNull\\(\(varName)\\)": "$1.isNull",
+            "CGRectIsEmpty\\(\(varName)\\)": "$1.isEmpty",
+            "CGRectIsInfinite\\(\(varName)\\)": "$1.isInfinite",
+            "CGRectStandardize\\(\(varName)\\)": "$1.standardized",
+            "CGRectIntegral\\(\(varName)\\)": "$1.integral",
+            "CGRectInset\\(\(varName),\(twoVariableOrNumber)\\)": "$1.insetBy(dx: $2, dy: $3)",
+            "CGRectOffset\\(\(varName),\(twoVariableOrNumber)\\)": "$1.offsetBy(dx: $2, dy: $3)",
+            "CGRectUnion\\(\(twoVars)\\)": "$1.union(rect: $2)",
+            "CGRectIntersection\\(\(twoVars)\\)": "$1.intersect(rect: $2)",
+            "CGRectContainsRect\\(\(twoVars)\\)": "$1.contains(rect: $2)",
+            "CGRectContainsPoint\\(\(twoVars)\\)": "$1.contains(point: $2)",
+            "CGRectIntersectsRect\\(\(twoVars)\\)": "$1.intersects(rect: $2)"
+            ]
+
+        let description = self.dynamicType.description
+        var corrections = [Correction]()
+        var contents = file.contents
+
+        for (pattern, template) in patterns {
+            let matches = file.matchPattern(pattern)
+                .filter({ $0.1.first == .Identifier })
+                .map({ $0.0 })
+
+            let regularExpression = regex(pattern)
+            for range in matches.reverse() {
+                contents = regularExpression.stringByReplacingMatchesInString(contents,
+                                options: [], range: range, withTemplate: template)
```

Comment:
That cause crash by running `swiftlint autocorrect --path test/618.swift`
`test/618.swift`:
```swift
CGRectGetWidth(rect)
CGRectGetHeight(rect)
CGRectGetMinX(rect)
CGRectGetMidX(rect)
CGRectGetMaxX(rect)
CGRectGetMinY(rect)
CGRectGetMidY(rect)
CGRectGetMaxY(rect)
CGRectIsNull(rect)
CGRectIsEmpty(rect)
CGRectIsInfinite(rect)
CGRectStandardize(rect)
CGRectIntegral(rect)
CGRectInset(rect, 10, 5)
CGRectOffset(rect, -2, 8.3)
CGRectUnion(rect1, rect2)
CGRectIntersection(rect1, rect2)
CGRectContainsRect(rect1, rect2)
CGRectContainsPoint(rect, point)
CGRectIntersectsRect(rect1, rect2)
```
```console
$ swiftlint autocorrect --config ~/.swiftlint-test.yml --path test/618.swift
Loading configuration from '/Users/norio/.swiftlint-test.yml'
Correcting Swift files at path test/618.swift
Correcting '618.swift' (1/1)
2016-04-17 09:31:19.104 swiftlint[24997:3063778] *** Terminating app due to uncaught exception 'NSRangeException', reason: '*** -[NSRegularExpression enumerateMatchesInString:options:range:usingBlock:]: Range or index out of bounds'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff9136d4f2 __exceptionPreprocess + 178
	1   libobjc.A.dylib                     0x00007fff8b4fef7e objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff913d44bd +[NSException raise:format:] + 205
	3   Foundation                          0x00007fff99e55f05 -[NSRegularExpression(NSMatching) enumerateMatchesInString:options:range:usingBlock:] + 347
	4   Foundation                          0x00007fff99e55d98 -[NSRegularExpression(NSMatching) matchesInString:options:range:] + 159
	5   Foundation                          0x00007fff99eed9ec -[NSRegularExpression(NSReplacement) stringByReplacingMatchesInString:options:range:withTemplate:] + 292
	6   SwiftLintFramework                  0x000000010d9d305d _TTSf4g_d___TFV18SwiftLintFramework29LegacyCGGeometryFunctionsRule11correctFilefC21SourceKittenFramework4FileGSaVS_10Correction_ + 16365
	7   SwiftLintFramework                  0x000000010d9c607f _TTWV18SwiftLintFramework29LegacyCGGeometryFunctionsRuleS_15CorrectableRuleS_FS1_11correctFilefC21SourceKittenFramework4FileGSaVS_10Correction_ + 15
	8   SwiftLintFramework                  0x000000010d9f85d6 _TFV18SwiftLintFramework6Linter7correctfT_GSaVS_10Correction_ + 502
	9   swiftlint                           0x000000010d9af4b8 swiftlint + 132280
	10  swiftlint                           0x000000010d9aed56 swiftlint + 130390
	11  swiftlint                           0x000000010d997950 swiftlint + 35152
	12  swiftlint                           0x000000010d9af10a swiftlint + 131338
	13  swiftlint                           0x000000010d9aee3f swiftlint + 130623
	14  swiftlint                           0x000000010d9af1be swiftlint + 131518
	15  Result                              0x000000010db8c94d _TFO6Result6Result8analysisurfT9ifSuccessFxqd__9ifFailureFq_qd___qd__ + 381
	16  Result                              0x000000010db8ddbb _TTWu0_R_s9ErrorTyperGO6Result6Resultxq__S0_10ResultTypeS0_FS2_8analysisurfT9ifSuccessFwx5Valueqd__9ifFailureFwx5Errorqd___qd__ + 235
	17  Result                              0x000000010db8afd1 _TFE6ResultPS_10ResultType7flatMapurfFwx5ValueGOS_6Resultqd__wx5Error_GS2_qd__wxS3__ + 145
	18  swiftlint                           0x000000010d9aea1a swiftlint + 129562
	19  swiftlint                           0x000000010d993ebd swiftlint + 20157
	20  Commandant                          0x000000010db53e03 _TFFV10Commandant14CommandWrappercuRd__S_11CommandTypexzWd__7Options11ClientError_wd__11ClientErrorzWd__S2_S3__rFqd__GS0_x_U_FCS_14ArgumentParserGO6Result6ResultT_GOS_15CommandantErrorQd____ + 2051
	21  Commandant                          0x000000010db532a4 _TPA__TFFV10Commandant14CommandWrappercuRd__S_11CommandTypexzWd__7Options11ClientError_wd__11ClientErrorzWd__S2_S3__rFqd__GS0_x_U_FCS_14ArgumentParserGO6Result6ResultT_GOS_15CommandantErrorQd____ + 132
	22  Commandant                          0x000000010db5194e _TFC10Commandant15CommandRegistry10runCommandfTSS9argumentsGSaSS__GSqGO6Result6ResultT_GOS_15CommandantErrorx___ + 318
	23  Commandant                          0x000000010db51e7a _TFC10Commandant15CommandRegistry4mainfT9argumentsGSaSS_11defaultVerbSS12errorHandlerFxT__T_ + 538
	24  Commandant                          0x000000010db51c57 _TFC10Commandant15CommandRegistry4mainfT11defaultVerbSS12errorHandlerFxT__T_ + 135
	25  swiftlint                           0x000000010d994edc swiftlint + 24284
	26  libdispatch.dylib                   0x00007fff97be993d _dispatch_call_block_and_release + 12
	27  libdispatch.dylib                   0x00007fff97bde40b _dispatch_client_callout + 8
	28  libdispatch.dylib                   0x00007fff97be229b _dispatch_root_queue_drain + 1890
	29  libdispatch.dylib                   0x00007fff97be1b00 _dispatch_worker_thread3 + 91
	30  libsystem_pthread.dylib             0x00007fff8b8544de _pthread_wqthread + 1129
	31  libsystem_pthread.dylib             0x00007fff8b852341 start_wqthread + 13
)
libc++abi.dylib: terminating with uncaught exception of type NSException
[1]    24997 abort      swiftlint autocorrect --config ~/.swiftlint-test.yml --path test/618.swift
```